lib/plugins/aws/provider/awsProvider.js

```diff
@@ -19,7 +19,7 @@ const constants = {
   providerName: 'aws',
 };
 
-const validAPIGatewayStageNamePattern = /^[a-zA-Z0-9_]+$/;
+const validAPIGatewayStageNamePattern = /^[-:a-zA-Z0-9_{}$.]+$/;
```

Comment:
I think this change is fine in most cases, but suggest a small enhancement.

It would be nice if stage name validation happens **after variable is resolved**, not when service definition parsing time like now.
Because it can validate a invalid stage name produced by `${variable}` too.

Reply:
@exoego, Thank you for comments.
Actually, because we are specifically talking about api gw stage name which is derived from provider.stage your sls validation will fail rather early before even reaching the api gw deployment, mainly due to stack creation. If I'm changing stage to an invalid value by referencing a variable I'm getting the following error:
```
 The stack service name "test-^hello" is not valid. A service name should only contain alphanumeric (case sensitive) and hyphens. It should start with an alphabetic character and shouldn't exceed 128 characters.
```

So nothing invalid will actually reach the api gw creation.

I do agree it would have been nice to run the check after the variable resolvement, but I have a feeling it's way more intrusive change, makes sense?

Reply:
Ok. @dschep what's next? Do you want to approve the pr?